service/src/test/java/bio/terra/pipelines/common/utils/StairwayFailedMetricsCounterHookTest.java

```diff
@@ -0,0 +1,142 @@
+package bio.terra.pipelines.common.utils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import bio.terra.pipelines.stairway.imputation.RunImputationJobFlightMapKeys;
+import bio.terra.pipelines.testutils.BaseEmbeddedDbTest;
+import bio.terra.pipelines.testutils.StairwayTestUtils;
+import bio.terra.pipelines.testutils.TestFlightContext;
+import bio.terra.stairway.FlightStatus;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class StairwayFailedMetricsCounterHookTest extends BaseEmbeddedDbTest {
+  StairwayFailedMetricsCounterHook stairwayFailedMetricsCounterHook =
+      new StairwayFailedMetricsCounterHook();
+
+  private SimpleMeterRegistry meterRegistry;
+
+  private final String meterRegistryName = "teaspoons.pipeline.failed.count";
+
+  @BeforeEach
+  void setup() {
+    meterRegistry = new SimpleMeterRegistry();
+    Metrics.globalRegistry.add(meterRegistry);
+    meterRegistry.clear();
+  }
+
+  @Test
+  void endFlight_notPipelineRunTypeFlight_success() throws InterruptedException {
```

Comment:
same comment about the camel casing and snake casing and smooshing these tests down into a smaller number.  These are a little more complicated but with some refactoring it can probably be made more manageable.  `@ParameterizedTest` might be useful here?